### PR TITLE
Branding update: Initial page structural changes

### PR DESCRIPTION
**Pull request by margaree**
Includes padding/margin changes, as well as having the component page top section have a different background color and bottom border.

The pilot banner was also removed. That change needed to happen anyways, and it was causing some weirdness with the background color for the top section of component pages.

### CODE REVIEW

File: .eleventy.cjs [hunk 1]
```diff
@@ -155,12 +178,6 @@ module.exports = function(eleventyConfig) {
 		} else return defaultTextRule(tokens, idx, options, env, slf);
 	};
 
-	const defaultHeading = markdownIt.renderer.rules.text;
```
Comment (margaree) [inline .eleventy.cjs:158]: There is already a `text` rule above (both did seem to be getting used before) so I just moved the check for hidden block above.

File: pages/_includes/css/style.css [hunk 2]
```diff
@@ -46,15 +52,29 @@ d2l-alert.d2l-daylight-warning-banner {
...
+	padding: 3.9rem 2.7rem 2.4rem 2.7rem;
+}
+.d2l-daylight-section-wrapper + .d2l-daylight-section-wrapper h2.d2l-heading-2,
+:not(.d2l-daylight-component-page) > .d2l-daylight-h1-section-wrapper + .d2l-daylight-section-wrapper h2.d2l-heading-2 {
+	border-top: 1px solid var(--d2l-color-gray-20);
+}
+.d2l-daylight-section-wrapper + .d2l-daylight-section-wrapper,
+:not(.d2l-daylight-component-page) > .d2l-daylight-h1-section-wrapper + .d2l-daylight-section-wrapper {
+	margin-top: 1.8rem;
+}
```
Comment (margaree) [inline pages/_includes/css/style.css:57]: We want the top h1 section of component pages to have a different background color that takes the full width of the content space and that also has a different border color than the border between other sections.

File: pages/_includes/layouts/base.njk [hunk 3]
```diff
@@ -184,23 +170,24 @@
 </head>
 <body class="d2l-typography">
 	<main class="d2l-daylight-content-unresolved">
-		<div class="d2l-daylight-side-panel">
-			<a href="/about/overview/" title="Home">
-				<div class="d2l-daylight-title">
-					<img src="/img/logo.svg" alt="Home" />
+		<div class="d2l-daylight-page-container ">
```
Comment (margaree) [inline pages/_includes/layouts/base.njk:173]: This is mostly just indentation. The `d2l-daylight-page-container` was added in order to be a flex container for the side nav and main content.

File: pages/_includes/layouts/base.njk [hunk 4]
```diff
@@ -184,23 +170,24 @@
...
+				{% set navPages = collections.all | eleventyNavigation | sortComponents %}
+				{% set toc = content | toc | safe %}
+				{% sideNav navPages, page.url, toc %}
 			</div>
-			{% set navPages = collections.all | eleventyNavigation | sortComponents %}
-			{% set toc = content | toc | safe %}
-			{% sideNav navPages, page.url, toc %}
-		</div>
-		<div class="d2l-daylight-content">
-			<d2l-alert class="d2l-daylight-warning-banner d2l-prevent-indexing" has-close-button type="default"><span style="font-size: 1.25rem; margin-right: .5rem;">🏗</span> This site is under construction and is available only as a pilot — see <d2l-link href="https://desire2learn.atlassian.net/wiki/spaces/GAUD/pages/2938569219/Piloting+the+New+Daylight+Site">Daylight Site Pilot</d2l-link>.</d2l-alert>
```
Comment (margaree) [inline pages/_includes/layouts/base.njk:201]: The alert was also removed

File: pages/_includes/layouts/component.njk [hunk 5]
```diff
@@ -20,22 +18,26 @@ permalink: "components/{{fileName}}/"
   window.addEventListener('hashchange', scrollToHash)
   window.addEventListener('DOMContentLoaded', scrollToHash)
 </script>
-<div class="d2l-prevent-indexing">
-  <h2 class="d2l-heading-2" id="future-enhancements" tab-index="-1">
-    <a class="d2l-header-anchor" href="#future-enhancements" aria-hidden="true">
+{% imports components %}
+
+<div class="d2l-daylight-component-page">
```
Comment (margaree) [inline pages/_includes/layouts/component.njk:23]: The changes here were mostly reorganization to (1) add the `d2l-daylight-component-page` container around the content, and (2) put the future enhancements section within a `d2l-daylight-section-wrapper` container in order to get the correct css rules.

File: .eleventy.cjs [hunk 6]
```diff
@@ -130,6 +130,28 @@ module.exports = function(eleventyConfig) {
...
+	// wrap content in section wrappers based on if in h1 or h2 section
+	markdownIt.renderer.rules.heading_open = (tokens, idx) => {
+		if (inHiddenBlock) return '';
+
+		const token = tokens[idx];
+		const tag = token.tag;
+
+		let result = '';
+		if (tag === 'h1') result = '<div class="d2l-daylight-h1-section-wrapper">';
+		else if (tag === 'h2') result = '</div><div class="d2l-daylight-section-wrapper">';
```
Comment (margaree) [inline .eleventy.cjs:142]: This does cause the need for a closing `</div>` after the last h2 section. I added one to the template for component pages. It is a bit weird.

**Review by dlockhart — APPROVED** [review]

**Pull request merged**